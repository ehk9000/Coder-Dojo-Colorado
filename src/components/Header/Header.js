import React from 'react';
import { connect } from 'react-redux';
import { NavLink, Link } from 'react-router-dom';
import PropTypes from 'prop-types';
import { ReactComponent as Logo } from '../../images/CoderDojoColorado.svg';
import { addUser } from '../../actions';

export function Header(props) {
  const logOutUser = () => {
    const { resetUser } = props;
    resetUser();
    localStorage.setItem('token', JSON.stringify(''));
  };

  const createMenuOptions = () => {
    const token = JSON.parse(localStorage.getItem('token'));
    let menuOptions;
    
    if (token) {
      menuOptions = <NavLink to="/" className="nav-link logout" onClick={logOutUser}>LOGOUT</NavLink>;
    } else {
      menuOptions = (
        <div>
          <NavLink to="/login" className="nav-link">LOGIN</NavLink>
          <NavLink className="nav-link" to="/register">REGISTER</NavLink>
        </div>
      );
    }

    return menuOptions;
  };

  return (
    <header>
      <Link to="/" className="logo-a-tag">
        <Logo className="logo" alt="Coder Dojo Colorado Logo" />
      </Link>
      <section>
        <NavLink className="nav-link" to="/">FIND A DOJO</NavLink>
        { createMenuOptions() }
      </section>
    </header>
  );
}

export const mapDispatchToProps = dispatch => ({
  resetUser: () => dispatch(addUser({}))
});

export default connect(null, mapDispatchToProps)(Header);

Header.propTypes = {
  resetUser: PropTypes.func
};

Header.defaultProps = {
  resetUser: () => {}
};
