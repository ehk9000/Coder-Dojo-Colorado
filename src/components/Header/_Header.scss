header {
  display: flex;
  flex-direction: row;
  align-items: center;
  justify-content: space-between;
}

header h1 {
  color:#49B749;
  margin-right: auto;
  align-self: center;
}

header a {
  text-decoration: none;
}

.logo-a-tag{
  display: inherit;
}

.logo {
  width: 175px;
  margin: 23px;
  margin-right: 2px;
}

.nav-link {
  margin-right: 55px;
  font-size: 18px;
  font-weight: 500;
  font-family: Lato;
  color:#000;
}

