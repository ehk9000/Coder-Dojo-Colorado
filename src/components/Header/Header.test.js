import React from 'react';
import { shallow } from 'enzyme';
import { Header, mapStateToProps } from './Header';

describe('Header', () => {
  let wrapper;
  const mockUser = {
    username: 'ehk',
    role: 0
  };

  beforeEach(() => {
    wrapper = shallow(<Header user={mockUser} />);
  });

  it('should match snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
