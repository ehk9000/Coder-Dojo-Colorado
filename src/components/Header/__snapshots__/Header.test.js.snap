// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header should match snapshot 1`] = `
<div>
  <h1>
     HEADER HEADER HEADER HEADER HEADER
  </h1>
</div>
`;
